### DISCUSSION TITLE
Crazy swarm, parallel task error

### DISCUSSION THREAD

**Discussion by fkorsah** [bitcraze/discussions#1570] [Q&A] [answered]
Hello, 

I am trying to test 2 crazyflies in a swarm using the Python code on the bitcraze website. But I keep on getting the error message as shown below. As of now, the code just tries to connect multiple drones and do a light check.
![Screenshot from 2024-10-01 13-11-35](https://github.com/user-attachments/assets/2bb105a7-0f29-4958-b2c9-7d65ca2b3739)

I saw there was a similar discussion that started 2 years ago but it did not solve my problem.
#326 
I have changed my crazyflies to fit my python script, and they do not share the same address.
Below is the script I am trying to run. 
```
import time

import cflib.crtp
from cflib.crazyflie.swarm import CachedCfFactory
from cflib.crazyflie.swarm import Swarm


def activate_led_bit_mask(scf):
    scf.cf.param.set_value('led.bitmask', 255)

def deactivate_led_bit_mask(scf):
    scf.cf.param.set_value('led.bitmask', 0)

def light_check(scf):
    activate_led_bit_mask(scf)
    time.sleep(2)
    deactivate_led_bit_mask(scf)
    time.sleep(2)


uris = {
    'radio://0/80/2M/E7E7E7E707',
    'radio://0/80/2M/E7E7E7E708',
    # Add more URIs if you want more copters in the swarm
}

if __name__ == '__main__':
    cflib.crtp.init_drivers()
    factory = CachedCfFactory(rw_cache='./cache')
    with Swarm(uris, factory=factory) as swarm:
         swarm.parallel_safe(light_check)
```

I do not know if I am missing some sort of driver or something, but I am able to fly a drone with a python script just fine, so I know I can connect to the drones (tested both URIs) using the crazyradio but the swarm throws up the error. And this code is from the website with a few tweaks to the uri.

I appreciate any help I can get regarding this issue.

**Comment by knmcguire** [comment]
Hi! I've put your code in proper code blocks for easier readability.

I've tried out your script exactly as it is but with different addresses, namely    ` 'radio://0/40/2M/E7E7E7E705',   'radio://0/40/2M/E7E7E7E702`'. It all worked out for me.

Could you tell anything else about your setup?

* Which cflib python version are you using?
* Which version of crazyradio are you using
* How far away are the crazyflies from the crazyradio?
* What is the native OS on your computer (from which you are running the script from.)

**Reply by fkorsah** [reply]
Hello.

I'm not sure how to check the version of cflib but I ran `pip3 install cflib` and got the result in the image
![Screenshot from 2024-10-02 11-29-08](https://github.com/user-attachments/assets/87482c5b-7899-4b1d-bae7-fd25e360d60f)

I am using the crazy radio 2.
The crazyflies are about one foot from the radio since I'm trying to do a light check.
I am running the script on Ubuntu 22.04.5

I also changed the URIs of my crazyflies to fit what you tested and I got the same error.

**Reply by knmcguire** [reply]
Some more questions:
- Are you using Crazyradio 1 or 2? 
- Which version of python are you using?

**Reply by fkorsah** [reply]
I am using the crazyradio 2.0 with Crazyflie 2.1+.
The version of python is Python 3.10.12

**Reply by jeremweb** [reply]
I encountered the exact same issue with the same logs, but on Windows 11 and Python 3.12. I managed to get the script working by changing the Crazyflies' addresses via the client, using radio://0/80/2M/E7E7E7E7E0 and radio://0/80/2M/E7E7E7E7E1. Since then, everything has been working perfectly.

**Reply by fkorsah** [reply, marked as answer]
Hi, I tried what you suggested and it worked. I then figured that technically the Crazyflie URI I had in the script and the actual URI were not the same. The crazyflie had a URI ending in E7 and the script had it ending in 07. Since the default URI is E7 and the script I copied from the website was 07. I didn't look at it much. But that's where the issue stemmed from. I just had to change the script to have the same 'E' instead of '0'. 
Thank you both for your help!

**Reply by knmcguire** [reply]
Ah yes that is indeed good you found it. The error `Exception: Too many packet lose` usually indicates it can't find one of the Crazyflies addresses (we need better error messages).
